Source/Core/VideoBackends/OGL/BoundingBox.cpp

```diff
@@ -2,69 +2,160 @@
 // Licensed under GPLv2+
 // Refer to the license.txt file included.
 
+#include <algorithm>
+#include <array>
 #include <cstring>
 
 #include "Common/GL/GLUtil.h"
 
 #include "VideoBackends/OGL/BoundingBox.h"
+#include "VideoBackends/OGL/FramebufferManager.h"
 
 #include "VideoCommon/DriverDetails.h"
 #include "VideoCommon/VideoConfig.h"
 
 static GLuint s_bbox_buffer_id;
 
+static std::array<int, 4> s_stencil_bounds{{0, 0, 0, 0}};
+static std::unique_ptr<u8[]> s_stencil_data;
+static bool s_stencil_updated;
+static bool s_stencil_cleared;
+
+static int s_target_width;
+static int s_target_height;
+
 namespace OGL
 {
-void BoundingBox::Init()
+void BoundingBox::SetEFBScaleChanged(float xscale, float yscale)
+{
+  if (g_ActiveConfig.backend_info.bSupportsFragmentStoresAndAtomics)
+    return;
+
+  s_target_width = static_cast<int>(EFB_WIDTH * xscale);
+  s_target_height = static_cast<int>(EFB_HEIGHT * yscale);
+  s_stencil_data = std::make_unique<u8[]>(s_target_width * s_target_height);
+  s_stencil_updated = false;
+}
+
+void BoundingBox::Init(float xscale, float yscale)
 {
-  if (g_ActiveConfig.backend_info.bSupportsBBox)
+  if (g_ActiveConfig.backend_info.bSupportsFragmentStoresAndAtomics)
   {
     int initial_values[4] = {0, 0, 0, 0};
     glGenBuffers(1, &s_bbox_buffer_id);
     glBindBuffer(GL_SHADER_STORAGE_BUFFER, s_bbox_buffer_id);
     glBufferData(GL_SHADER_STORAGE_BUFFER, 4 * sizeof(s32), initial_values, GL_DYNAMIC_DRAW);
     glBindBufferBase(GL_SHADER_STORAGE_BUFFER, 0, s_bbox_buffer_id);
   }
+  else
+  {
+    SetEFBScaleChanged(xscale, yscale);
+  }
 }
 
 void BoundingBox::Shutdown()
 {
-  if (g_ActiveConfig.backend_info.bSupportsBBox)
+  if (g_ActiveConfig.backend_info.bSupportsFragmentStoresAndAtomics)
+  {
     glDeleteBuffers(1, &s_bbox_buffer_id);
+  }
+  else
+  {
+    s_stencil_data.reset();
+  }
 }
 
 void BoundingBox::Set(int index, int value)
 {
-  glBindBuffer(GL_SHADER_STORAGE_BUFFER, s_bbox_buffer_id);
-  glBufferSubData(GL_SHADER_STORAGE_BUFFER, index * sizeof(int), sizeof(int), &value);
-  glBindBuffer(GL_SHADER_STORAGE_BUFFER, 0);
+  if (g_ActiveConfig.backend_info.bSupportsFragmentStoresAndAtomics)
+  {
+    glBindBuffer(GL_SHADER_STORAGE_BUFFER, s_bbox_buffer_id);
+    glBufferSubData(GL_SHADER_STORAGE_BUFFER, index * sizeof(int), sizeof(int), &value);
+    glBindBuffer(GL_SHADER_STORAGE_BUFFER, 0);
+  }
+  else
+  {
+    s_stencil_bounds[index] = value;
+
+    if (!s_stencil_cleared)
+    {
+      GLint old_fbo;
+      glGetIntegerv(GL_DRAW_FRAMEBUFFER_BINDING, &old_fbo);
+
+      glBindFramebuffer(GL_FRAMEBUFFER, FramebufferManager::GetEFBFramebuffer());
+      glClearStencil(0.0f);
+      glClear(GL_STENCIL_BUFFER_BIT);
```

Comment:
You could readback the other three values before, if there was any update in between. This should also not happen in the common use case as well.